### PR TITLE
For #19055: Implement ConstraintLayout

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by kglazko** [comment]
Screen Shot 2021-04-20 at 11 48 25 AM
Screen Shot 2021-04-20 at 11 48 33 AM

**Comment by gabrielluong** [comment]
We got a regression on how the expiration date is suppose to look here. It is expected that they take up the full horizontal length and have an underline below each of the expiry year and month. The expiry year needs to be a bit dynamic in terms of its to sizing to account for all the different years and locale.

We should also try to remove the `android:id` we don't need or those that don't follow the naming convention which uses underscores(_).

File: app/src/main/res/layout/fragment_credit_card_editor.xml [hunk 1]
```diff
@@ -94,86 +101,62 @@
...
-            <androidx.appcompat.widget.AppCompatSpinner
-                android:id="@+id/expiry_month_drop_down"
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                tools:listitem="@android:layout/simple_list_item_1" />
 
-            <View
-                android:layout_width="match_parent"
-                android:layout_height="1dp"
-                android:background="?primaryText" />
```
Comment (jonalmeida) [inline app/src/main/res/layout/fragment_credit_card_editor.xml:128]: Leaving a note here, not to forget to add the view lines back in based on our offline conversation.

**Comment by gabrielluong** [comment]
Closing this since we aren't pursuing it at this time.

**Pull request closed without merging**